### PR TITLE
boards: arm: add OtterPill STM32F072 board

### PR DESCRIPTION
**Pull request by ottojo**
Add support for https://github.com/Jana-Marie/OtterPill board

* USB-C with PD support (FUSB302B)
* STM32F072X8
* Arduino Nano form factor (compatible pinout, but 3.3V logic)

docs: https://builds.zephyrproject.io/zephyr/pr/62830/docs/boards/arm/otterpill/doc/index.html

### CODE REVIEW

File: boards/arm/otterpill/otterpill.dts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+&i2c1 {
+	status = "okay";
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+};
+
+&i2c2 {
+	// Board-internal I2C bus, connects to FUSB302B
+	// (for which there exists no Zephyr driver yet)
```
Comment (erwango) [inline boards/arm/otterpill/otterpill.dts:120]: Please use C style comments

File: boards/arm/otterpill/otterpill.dts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		};
+	};
+
+	aliases {
+		led0 = &led_status;
+		sw0 = &dfu_button;
+		watchdog0 = &iwdg;
+	};
+
+	arduino_nano_header: connector {
```
Comment (erwango) [inline boards/arm/otterpill/otterpill.dts:43]: Please use a dedicated arduino header dtsi file. See nucleo boards for instance.

File: boards/arm/otterpill/otterpill.dts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ */
+
+/dts-v1/;
+#include <st/f0/stm32f072X8.dtsi>
+#include <st/f0/stm32f072c(8-b)tx-pinctrl.dtsi>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "OtterPill";
+
```
Comment (erwango) [inline boards/arm/otterpill/otterpill.dts:13]: Please add a board compatible

File: boards/arm/otterpill/otterpill.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+type: mcu
+arch: arm
+ram: 16
+flash: 64
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+supported:
+  - gpio
```
Comment (erwango) [inline boards/arm/otterpill/otterpill.yaml:12]: watchdog, spi, usb_device

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for contributing this board. First round of review. I'll need to dig the DFU part.

File: boards/arm/otterpill/otterpill.dts (hunk 1 above)
Comment (ottojo) [reply boards/arm/otterpill/otterpill.dts]: done

File: boards/arm/otterpill/otterpill.dts (hunk 2 above)
Comment (ottojo) [reply boards/arm/otterpill/otterpill.dts]: done

File: boards/arm/otterpill/otterpill.dts (hunk 3 above)
Comment (ottojo) [reply boards/arm/otterpill/otterpill.dts:15]: done

File: boards/arm/otterpill/otterpill.yaml (hunk 4 above)
Comment (ottojo) [reply boards/arm/otterpill/otterpill.yaml:12]: done

File: boards/arm/otterpill/otterpill.dts (hunk 1 above)
Comment (ottojo) [reply boards/arm/otterpill/otterpill.dts]: @erwango now that the merge window is open again, please re-review, thanks!

**Comment by erwango** [comment]
Ok, I've just seen that this your own personal board.
I haven't seen specific written policy around this but it is not obvious that such specific board support can be merged in tere. @MaureenHelm, @galak thoughts ?

**Comment by ottojo** [comment]
> Ok, I've just seen that this your own personal board. I haven't seen specific written policy around this but it is not obvious that such specific board support can be merged in tere. @MaureenHelm, @galak thoughts ?

Not my own, but just sold through a tindie store, not commercially available through the usual vendors. I understand only wanting to have boards with proper availability and support in tree, if that's the case I will just make an OOT module, which isn't really less convenient than in-tree for anyone!

**Comment by ottojo** [comment]
realized out of tree at https://github.com/ottojo/otterpill_zephyr

**Pull request closed without merging**